### PR TITLE
refactor ban command

### PR DESCRIPTION
**Pull request by JulianVennen**
- adds the class ModerationCommand which implements shared code of all moderation commands
- adds TimedModerationCommand for moderation commands with a duration

### CODE REVIEW

File: src/Guild.js [hunk 1]
```diff
@@ -0,0 +1,48 @@
+const RateLimiter = require('./RateLimiter');
+
+class Guild {
```
Comment (matthi4s) [inline src/Guild.js:3]: Why are all functions static? You could create guild objects with the guild id etc., optionally with a get function and some form of cache.

File: src/commands/legacy/strike.js [hunk 2]
```diff
@@ -147,7 +147,18 @@ command.executePunishment = async (punishment, guild, user, bot, database, reaso
...
   switch (punishment.action) {
     case 'ban':
-      await ban.ban(guild, user, bot.user, reason, punishment.duration);
+      const ban = new Ban(/** @type {module:"discord.js".Message}*/{
+        guild,
+        author: bot.user,
+        content: '---',
+      }, database, bot, 'ban');
+      ban.guildConfig = await GuildConfig.get(/** @type {module:"discord.js".Snowflake} */guild.id);
+      ban.reason = reason; //DURATION DNW
```
Comment (matthi4s) [inline src/commands/legacy/strike.js:156]: What does that comment mean?

File: src/commands/legacy/strike.js [hunk 3]
```diff
@@ -147,7 +147,18 @@ command.executePunishment = async (punishment, guild, user, bot, database, reaso
   let member;
   switch (punishment.action) {
     case 'ban':
-      await ban.ban(guild, user, bot.user, reason, punishment.duration);
+      const ban = new Ban(/** @type {module:"discord.js".Message}*/{
```
Comment (matthi4s) [inline src/commands/legacy/strike.js:150]: You should probably separate the Ban itself from the command. "Faking" a command call is not really a clean solution.

**Review by matthi4s — CHANGES_REQUESTED** [review]

File: src/commands/ban.js [hunk 4]
```diff
@@ -1,14 +1,14 @@
-const util = require('../../util.js');
-const Log = require('../../Log');
-const GuildConfig = require('../../GuildConfig');
-const RateLimiter = require('../../RateLimiter');
+const util = require('../util.js');
```
Comment (matthi4s) [inline src/commands/ban.js:1]: Shouldn't this file be deleted?

File: src/Member.js [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+    }
+
+    /**
+     * send the user a dm about this punishment
+     * @param {String}  verb
+     * @param {String}  reason
+     * @param {Number}  duration
+     * @return {Promise<Boolean>} success
+     */
+    async dmUser(verb, reason, duration) {
```
Comment (matthi4s) [inline src/Member.js:48]:
```suggestion
    async dmPunishedUser(verb, reason, duration) {
```

**Review by matthi4s — CHANGES_REQUESTED** [review]

**Review by KurtThiemann — APPROVED** [review]

**Review by matthi4s — APPROVED** [review]

**Pull request merged**